spock-core/src/main/java/org/spockframework/runtime/ValueRecorder.java

```diff
@@ -27,6 +27,7 @@
  */
 public class ValueRecorder {
   private final ArrayList<Object> values = new ArrayList<Object>();
+  private final Stack<Integer> startedRecordings = new Stack<Integer>();
```

Comment:
Why use `Stack` here? I should be `Deque<Integer> startedRecordings= new ArrayDeque<Integer>();`